test/e2e/page-objects/pages/header-navbar.ts

```diff
@@ -63,6 +65,20 @@ class HeaderNavbar {
     await this.driver.clickElement(this.settingsButton);
   }
 
+  async clickSwitchNetworkDropDown(): Promise<void> {
+    console.log(`Click switch network menu`);
+    await this.driver.clickElement(this.switchNetworkDropDown);
+  }
+
+  async check_networkNameSwitchDropDown(networkName: string): Promise<boolean> {
+    console.log(`Validate the Switch network to ${networkName}`);
+    const switchNetworkName = await this.driver.findElements({
+      tag: 'span',
+      text: networkName,
+    });
+    return switchNetworkName.length === 1;
```

Comment:
could you explain why we need to compare the length here? can we just check the current selected network by 
await this.driver.waitForSelector({
      tag: 'span',
      text: networkName,
    });
 Also, i feel this method name is not clear enough, can we changes it to something like `check_currerntSelectedNetwork`

Reply:
ok thanks @chloeYue, I will make the change.

Reply:
@chloeYue Let me explain why I used this format. Actually, assertions in the page model dont happen. Please refer to the documentation provided by Selenium under the section “Assertions in Page Object” [here](https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/). Hence, I had return statement, but I missed the assert statement. So, it's a good catch from you!

After modifying the code with the suggestion you provided, if I give any invalid drop name, it would not catch correctly. Therefore, I have modified it to this pattern, and it works fine.

```
await this.driver.waitForSelector(
      `button[data-testid="network-display"][aria-label="Network Menu ${networkName}"]`,
    );
```

Also, I believe you wanted the function name(check_currentSelectedNetwork) to be this, right? Let me know if there any other feedback or questions.

Reply:
Thanks for the change, looks good to me !